examples/github_actions/sample_repo/launch_sky.py

```diff
@@ -0,0 +1,120 @@
+"""Launch a SkyPilot task with custom resources."""
+
+import argparse
+import copy
+import logging
+import os
+import sys
+from typing import Optional
+
+import sky
+from sky.jobs.client import sdk as jobs_sdk
+
+logger = logging.getLogger('launch.py')
+
+
+def patch_task(
+    task: sky.Task,
+    cpus: Optional[int] = None,
+    gpus: Optional[int] = None,
+    nodes: Optional[int] = None,
+) -> sky.Task:
+    """Patch the task to add the requested resources overrides."""
+    overrides = {}
+    if cpus:
+        overrides['cpus'] = cpus
+    if overrides:
+        task.set_resources_override(overrides)
+    if nodes:
+        task.num_nodes = nodes
+
+    new_resources_list = list(task.resources)
+
+    if gpus:
+        new_resources_list = []
+        for res in list(task.resources):
+            if not res.accelerators:
+                # No accelerators, so no need to patch
+                continue
+            if not isinstance(res.accelerators, dict):
+                # shouldn't happen with our current config
+                raise Exception('Unexpected accelerator type: '
+                                f'{res.accelerators}, {type(res.accelerators)}')
+
+            patched_accelerators = copy.deepcopy(res.accelerators)
+            patched_accelerators = {
+                gpu_type: gpus for gpu_type in patched_accelerators.keys()
+            }
+            new_resources = res.copy(accelerators=patched_accelerators)
+            new_resources_list.append(new_resources)
+
+    if gpus:
+        task.set_resources(type(task.resources)(new_resources_list))
+
+    return task
```

Comment:
I don't see a significant benefit for using the SDK to override the resources here, and it is also quite complicated. I think we can get rid of this. For the benefit of being able to return the request_id, I am a bit hesitating for a user to create a python script for that. An alternative way is that in some other systems (such as GCP, or AWS) the CLI may be able to return some json output, so people can parse it programmatically